ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -220,72 +152,117 @@ void IrrlichtBaWTexture::loadFromMemory(const void* buffer, const CEGUI::Sizef&
     region.imageSubresource.layerCount = 1u;
     region.bufferOffset = 0u;
     region.bufferRowLength = 0;
-    region.bufferImageHeight = 0u; 
+    region.bufferImageHeight = 0u;
     region.imageOffset = { 0u, 0u, 0u };
     region.imageExtent = imageParams.extent;
 
-    auto gpuImage = device->getVideoDriver()->createGPUImage(std::move(imageParams));
-    driver->copyBufferToImage(gpuPixelBuffer.get(), gpuImage.get(), regions->size(), regions->begin());
+    driver->copyBufferToImage(gpuBuffer.get(), gpuImage.get(), regions->size(), regions->begin());
```

Comment:
first create the GPU Image, then do a `copyBufferToImage` but via the upstreaming staging buffer, don't create new buffers!

Reply:
also DRY, so create the IGPUImage and the views, then call `blitFromMemory`